data_steward/tools/snapshot_by_query.py

```diff
@@ -20,13 +27,33 @@ def create_empty_dataset(project_id, dataset_id, snapshot_dataset_id):
         overwrite_existing=True)
 
 
-def create_empty_cdm_tables(snapshot_dataset_id):
+def has_at_birth_column(dataset_id):
+    query = """SELECT column_name from `{dataset}.INFORMATION_SCHEMA.COLUMNS` where table_name = 'person' """
+    client = bq.Client()
+    result = client.query(query.format(dataset=dataset_id)).to_dataframe()
+    if 'sex_at_birth_concept_id' in result['column_name'].to_list():
+        return True
+    else:
+        return False
+
+
+def create_empty_cdm_tables(snapshot_dataset_id, dataset_id):
     """
     Copy the table content from the current dataset to the snapshot dataset
     :param snapshot_dataset_id:
     :return:
     """
-    cdm.create_all_tables(snapshot_dataset_id)
+    for table in resources.CDM_TABLES:
+        if table == PERSON and has_at_birth_column(dataset_id):
+            table_id = table
+            table_name = 'post_deid_person'
+        else:
+            table_id = table
+            table_name = table
+        create_standard_table(table_name,
+                              table_id,
+                              drop_existing=True,
+                              dataset_id=snapshot_dataset_id)
```

Comment:
nit: This might be simpler
```suggestion
    for table in resources.CDM_TABLES:
        table_id = table
        table_name = table
        if table == PERSON and has_at_birth_column(dataset_id):
            table_name = 'post_deid_person'
        create_standard_table(table_name,
                              table_id,
                              drop_existing=True,
                              dataset_id=snapshot_dataset_id)
```